sparse/_coo/core.py

```diff
@@ -2186,6 +2186,9 @@ def astype(self, dtype, copy=True):
         :obj:`COO.elemwise`: Apply an arbitrary element-wise function to one or two
             arguments.
         """
+        # this matches numpy's behavior
+        if self.dtype == dtype and copy == False:
```

Comment:
Comparing booleans to booleans is a weird pattern.

```suggestion
        if self.dtype == dtype and not copy:
```

Reply:
This makes `coo.astype(int, copy=[])` succeed, whereas the way I've written it it gives `TypeError: an integer is required (got type list)`

Reply:
I'd still prefer that, anyone who writes that (or equivalent) code is in much hotter waters.

Reply:
Feel free to commit this change yourself if you feel that way. I'd usually lean towards preserving numpy errors when possible, simply because it makes it easier for users to spot typos.